### PR TITLE
Updated to latest GitVersionTask v3

### PR DESCRIPTION
**Pull request by andreasohlund**
This was needed to configure GV to bump to "RC"

### CODE REVIEW

File: GitVersionConfig.yaml [hunk 1]
```diff
@@ -3,3 +3,5 @@ next-version: 6.0
 branches:
   develop:
     tag: alpha
+  releases?[/-]:
```
Comment (andreasohlund) [inline GitVersionConfig.yaml]: @SimonCropp @Particular/nservicebus-maintainers I think this is a resonable default going forwards => will make minor releases be "RC" status right away?

This way you only have to change this when going into a new major (change it down to `beta`)

@JakeGinnivan is there a way to configure GV to say "Major releases should start with `beta` but minor/patch should start with `RC`" ?

Comment (JakeGinnivan) [inline GitVersionConfig.yaml]: I have been thinking about that. I have an open issue at https://github.com/GitTools/GitVersion/issues/1018 if you want to jump in and suggest that :)

Comment (JakeGinnivan) [inline GitVersionConfig.yaml]: Also, v4 is well underway. It fixes a bunch of commit counting issues and it changes the default tags from unstable to alpha but bumps things in a different way.

Comment (andreasohlund) [inline GitVersionConfig.yaml]: Thanks @jakeginnivan, chimed in: https://github.com/GitTools/GitVersion/issues/1018#issuecomment-243085068

**Comment by bording** [comment]
The `GitVersionConfig.yaml` file name should be renamed to `GitVersion.yml` because the old name has been obsoleted.

Also, looks like 3.6.3 just came out https://github.com/GitTools/GitVersion/releases/tag/v3.6.3, so we should probably go ahead and bump this PR up to that one instead.

**Comment by andreasohlund** [comment]
@bording should be good to go now

**Comment by johnsimons** [comment]
@andreasohlund @bording so if I remember correctly the latest of GV did not stamp the date on our binaries, is this now fixed ?

**Comment by bording** [comment]
@johnsimons Based on this change: https://github.com/Particular/NServiceBus/pull/4094, I don't think so.

**Comment by johnsimons** [comment]
@bording so is it safe to be using GV v3 ? Are you guys planning to maintain the release date manually ?

**Comment by bording** [comment]
@johnsimons This PR doesn't introduce v3, we've already been on it for a while. I believe the plan is to manually maintain it, until we get some new automated solution.

**Comment by johnsimons** [comment]
> This PR doesn't introduce v3, we've already been on it for a while

I know


> I believe the plan is to manually maintain it, until we get some new automated solution.

good luck!

**Comment by JakeGinnivan** [comment]
Just to let you all know, v4 is well under way. It changes commit counting behavior and develop now uses the alpha tag, but as soon as a release branch is created it bumps develop.

**Comment by JakeGinnivan** [comment]
https://github.com/GitTools/GitVersion/releases/tag/v4.0.0-beta.1 for more info. Docs are slightly out of date, few things have changed

**Comment by andreasohlund** [comment]
@johnsimons yea, not optimal but we'll get it done https://github.com/Particular/Operations.Licensing/issues/16

**Pull request merged**